### PR TITLE
[MRG] Change filter behaviour to return all objects if no filter parameters are specified

### PR DESCRIPTION
**Pull request by JuliaSprenger**
Closes issue #493

This allows to filter for objects of a specific type without specifying another filter attribute.
Previously filters like ```targdict={}, object=neo.SpikeTrain``` returned an empty list, where as now this returns all SpikeTrain objects available.

### CODE REVIEW

**Comment by JuliaSprenger** [comment]
Failing tests are not related to this PR and will be fixed in PR #496, which should be merged first.

**Comment by samuelgarcia** [comment]
You can re run tests now.

**Comment by samuelgarcia** [comment]
I must confess that I do not use *filterdata* so I was not aware of the old behavior.

I will put some coment.

The first impression: the doc is missing a bit for the why target and/or **kargs.
And also targetdict=None meaning *no filter* should be more explicit.

But this do not come from your patch.

File: neo/core/container.py [hunk 1]
```diff
@@ -56,27 +56,27 @@ def filterdata(data, targdict=None, objects=None, **kwargs):
         targdict += [kwargs]
 
     if not targdict:
```
Comment (samuelgarcia) [inline neo/core/container.py]: I prefer the more explicit:
if targdict is not None:

File: neo/core/container.py [hunk 2]
```diff
@@ -56,27 +56,27 @@ def filterdata(data, targdict=None, objects=None, **kwargs):
...
-        return []
+        results = data
 
     # if multiple dicts are provided, apply each filter sequentially
-    if not hasattr(targdict, 'keys'):
+    elif not hasattr(targdict, 'keys'):
```
Comment (samuelgarcia) [inline neo/core/container.py]: Maybe this would be more explcit:
```elif isinstance(targdict, list):```

**Comment by samuelgarcia** [comment]
Except improving a bit the doc, I think this is OK for merging.
The two coments I made are useless.

**Comment by JuliaSprenger** [comment]
I extended the documentation. Ready to merge?

**Pull request merged**